src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -493,4 +511,56 @@ public static Object coerceBodyType(InputStream body, ResponseBodyType as,
 
         return response;
     }
+
+    private static Timer.Context startTimer(MetricRegistry registry, HttpRequest request) {
+        if (registry != null) {
+            final RequestLine requestLine = request.getRequestLine();
+            final String name = MetricRegistry.name("puppetlabs.http-client", requestLine.getUri(),
```

Comment:
Can we extract a constant for `puppetlabs.http-client`?

Also it looks like we're just using the raw URI string for the metrics names.  Did you have thoughts on how that maps to the 'metric-id' tuples that we discussed earlier?  I think that working the code up to that point incrementally is fine, but if we wanted to merge this chunk of work before those changes landed, I think we'd need to have a pretty clear idea of how we might evolve it to add those.  Perhaps it could be as simple as adding another namespace to the metric name in these code paths, e.g. `puppetlabs.http-client.unidentified-requests` or `raw-requests` or something.